Lab_1-2-3/lab-1-2-3-sql-window-function.md

```diff
@@ -995,80 +1026,118 @@ where productid = 1 and year (date) = 2022
+WHERE productid = 1 AND year(date) = 2022
 ORDER BY date;
 ```
 
-**Wyniki**
+**Wyniki zapytań**
 
-| MS SQL                     | Postgres                    | SQLite                    |
-| -------------------------- | --------------------------- | ------------------------- |
-| ![](./img/ex10/mysql1.png) | ![](img/ex10/postgres1.png) | ![](img/ex10/sqlite1.png) |
+| Zapytanie | MS SQL                     | Postgres                    | SQLite                    |
+| --------- | -------------------------- | --------------------------- | ------------------------- |
+| 1         | ![](./img/ex10/mssql1.png) | ![](img/ex10/postgres1.png) | ![](img/ex10/sqlite1.png) |
+| 2         | ![](./img/ex10/mssql2.png) | ![](img/ex10/postgres2.png) | ![](img/ex10/sqlite2.png) |
 
-**Plan wykonania**
+**Czasy wykonania**
 
-| MS SQL                         | Postgres                        | SQLite                          |
-| ------------------------------ | ------------------------------- | ------------------------------- |
-| ![](./img/ex10/mysql1plan.png) | ![](img/ex10/postgres1plan.png) | ![](./img/ex10/sqlite1plan.png) |
+| Zapytanie | MS SQL | Postgres | SQLite |
+| --------- | ------ | -------- | ------ |
+| 1         | 2.4 s  | 1.5 s    | 1.2 s  |
+| 2         | 2.3 s  | 1.7 s    | 1.0 s  |
 
-**Czasy wykonania**
+**Plany wykonania**
 
-| MS SQL                         | Postgres                        | SQLite                          |
-| ------------------------------ | ------------------------------- | ------------------------------- |
-| ![](./img/ex10/mysql1time.png) | ![](img/ex10/postgres1time.png) | ![](./img/ex10/sqlite1time.png) |
+| Zapytanie | MS SQL                         | Postgres                        | SQLite                          |
+| --------- | ------------------------------ | ------------------------------- | ------------------------------- |
+| 1         | ![](./img/ex10/mssql1plan.png) | ![](img/ex10/postgres1plan.png) | ![](./img/ex10/sqlite1plan.png) |
+| 2         | ![](./img/ex10/mssql2plan.png) | ![](img/ex10/postgres2plan.png) | ![](./img/ex10/sqlite2plan.png) |
 
-> Zostały uzyskane takie same wyniki po wykonaniu funkcji.
+> W tym zadaniu, konieczne było skorzystanie z tabeli `product_history`, która posiada 2500 rekordów.
 
-> Z definicji oraz na podstawie powyższych wyników można wywnioskować, że funkcja LAG umożliwia dostęp do wartości z poprzedniego wiersza w ramach określonego zestawu danych.
-> Funkcja LEAD natomiast umożliwia dostęp do wartości z następnego wiersza w ramach określonego zestawu
+> Zapytania z funkcjami okna wykonują się znacznie szybciej, są prostrze, czytelniejsze i bardziej zwięzłe. Zapytania bez funkcji okna są bardziej skomplikowane, wymagają złączeń oraz podzapytań, co sprawia, że są mniej czytelne i trudniejsze do zrozumienia. Zapytania z użyciem funkcji okna mają również wielokrotnie mniejszy koszt wykonania i prostszy plan wykonania niż ich odpowiedniki bez funkcji okna.
```

Comment:
Może pomińmy to, że są trudniejsze do zrozumienia? Sama słaba czytelność wystarczy

Reply:
Usunąłem "i trudniejsze do zrozumienia"